python/packages/autogen-core/src/autogen_core/_single_threaded_agent_runtime.py

```diff
@@ -830,6 +831,33 @@ async def factory_wrapper() -> T:
 
         return type
 
+    async def register_agent_instance(
+        self,
+        agent_instance: T | Awaitable[T],
+        agent_id: AgentId,
+    ) -> AgentId:
+        def agent_factory() -> T:
+            raise RuntimeError("Agent factory should not be called when registering an agent instance.")
```

Comment:
Can you make this error clearer? This is what a user will see if they broadcast something and it hits a non-existent instance right? It's a good place to educate on how to handle (i.e. using skip_class_subscriptions if it was unintended)